drivers/serial/uart_stm32.c

```diff
@@ -896,6 +896,9 @@ static void uart_stm32_isr(const struct device *dev)
 #ifdef CONFIG_PM
 		uart_stm32_pm_constraint_release(dev);
 #endif
+	} else if (LL_USART_IsEnabledIT_RXNE(UartInstance)) {
```

Comment:
```suggestion
	} else if (LL_USART_IsActiveFlag_RXNE(UartInstance)) {
```
LGTM, but I wonder if `LL_USART_IsActiveFlag_RXNE()` will be more appropriate?  If I understand it correctly, we need to flush only when there this something inside HW FIFO

Reply:
not wrong, If we reach this code of the uart_stm32_isr interrupt, maybe even better is to combine :
```
	} else if (LL_USART_IsEnabledIT_RXNE(UartInstance) &&
			  LL_USART_IsActiveFlag_RXNE(UartInstance)) {

```

Reply:
I concur.

Reply:
done